core/src/main/scala/kafka/api/ControlledShutdownRequest.scala

```diff
@@ -58,7 +58,7 @@ case class ControlledShutdownRequest(versionId: Short,
   def sizeInBytes: Int = {
     2 + /* version id */
       4 + /* correlation id */
-      clientId.fold(0)(shortStringLength)
+      clientId.fold(0)(shortStringLength) +
```

Comment:
Oh, I understand. It's because we use the Java implementation of `ControlledShutdownRequest` unless "inter.broker.protocol.version" is < 0.9.0.0. So, we would only have caught this via the (in-progress) upgrade tests.